### ISSUE TITLE
How to implement MaterialSearchView in a fragment?

### ISSUE THREAD

**Issue by LuizHonorato** [Mauker1/MaterialSearchView#96]
Good evening, I'm try to implement this project in a fragment, but without sucess. Code:

`public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        setHasOptionsMenu(true);

        return inflater.inflate(R.layout.fragment_buscar, container, false);
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        super.onCreateOptionsMenu(menu, inflater);
        menu.clear();
        inflater.inflate(R.menu.campo_pesquisa, menu);

            MaterialSearchView searchView = (MaterialSearchView) menu.findItem(R.id.barra_pesquisa);

            searchView.setOnQueryTextListener(new MaterialSearchView.OnQueryTextListener() {
                @Override
                public boolean onQueryTextSubmit(String s) {
                    return false;
                }

                @Override
                public boolean onQueryTextChange(String s) {
                    return false;
                }
            });

        searchView.setSearchViewListener(new MaterialSearchView.SearchViewListener() {
            @Override
            public void onSearchViewOpened() {

            }

            @Override
            public void onSearchViewClosed() {

            }
        });

    }
`
Someone can help me please?

**Comment by Mauker1** [comment]
You should handle the Search View on the Activity. If your Fragment needs the results you have a few options on how to send it to the fragment, such as:

- Broadcast the information via [`LocalBroadcastManager`](https://developer.android.com/reference/android/support/v4/content/LocalBroadcastManager.html)

- Send and event using an [EventBus](greenrobot.org/eventbus/documentation/).

- Use a callback method.

**Comment by AdamMc331** [comment]
I agree with Mauker on the solutions above, but I do think adding support for Fragment use would be beneficial. 

Can you explain more why you were unsuccessful in implementing it in your Fragment? What was the issue with the code above?

**Comment by Faiyyaz** [comment]
java.lang.NullPointerException: Attempt to invoke virtual method 'void br.com.mauker.materialsearchview.MaterialSearchView.setOnQueryTextListener(br.com.mauker.materialsearchview.MaterialSearchView$OnQueryTextListener)' on a null object reference

Facing this error while adding in fragment

**Comment by Mauker1** [comment]
Is the search view on your fragment layout? Are you sure you're getting its reference once the view is inflated?

**Comment by Faiyyaz** [comment]
Its a menu item and yes i m taking it reference as u can see in the above mentioned code snippet

**Comment by Mauker1** [comment]
Try to find the MSV on your `onCreateView` method.

**Comment by Faiyyaz** [comment]
I have used it as a menu item on top of action bar i don't want to use it inside my fragment because i have 3 fragments where i want search so i want it to be a common part of the app
